pkg/monitoring/vmistats/collector.go

```diff
@@ -104,12 +138,49 @@ func (co *VMICollector) Collect(ch chan<- prometheus.Metric) {
 		vmis[i] = obj.(*k6tv1.VirtualMachineInstance)
 	}
 
-	updateVMIsPhase(vmis, ch)
+	updateVMIsPhase(vmis, co.buildVmisInstanceTypes(), ch)
 	co.updateVMIMetrics(vmis, ch)
 	return
 }
 
-func (vmc *vmiCountMetric) UpdateFromAnnotations(annotations map[string]string) {
+func (co *VMICollector) buildVmisInstanceTypes() *vmisInstanceTypesVendors {
+	cachedInstanceTypes := co.instanceTypeInformer.GetIndexer().List()
```

Comment:
Actually can we just pass the informers down instead of creating the map?

Reply:
and get each instance type individually?
although smaller we would be making many more requests, no?

Reply:
More requests to the local cache yes but that shouldn't impact the number of external calls.

Reply:
updated